src/ol/reproj.js

```diff
@@ -330,7 +330,7 @@ export function render(
     context.save();
     context.beginPath();
 
-    if (isBrokenDiagonalRendering()) {
+    if (isBrokenDiagonalRendering() || !isEmpty(opt_contextOptions)) {
```

Comment:
You could make this more specific and check for the imageSmooting option, instead of just checking whether there are contextOptions. One thing less to worry about in the future, in case other options get added.

Reply:
That is going to need some rework anyway.  It seems the latest version of Chrome behaves unexpectedly if `msImageSmoothingEnabled` (which is still needed by IE) is set so instead of passing an object containing both there will need to be browser specific code.